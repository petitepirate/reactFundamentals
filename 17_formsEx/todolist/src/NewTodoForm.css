input {
	display: block;
	margin: 10px auto;
	width: 300px;
}

.add-btn {
	margin-bottom: 50px;
}
