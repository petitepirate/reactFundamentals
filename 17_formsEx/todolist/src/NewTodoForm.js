import React from 'react';
import './NewTodoForm.css'

function NewTodoForm({ formData, changeHandler, submitHandler }) {
  return (
  <form onSubmit={submitHandler}>
    <label htmlFor="todo">Input items for your to-do list:</label>
    <input 
    type="text"
    id="todo"
    name="todo"
    value={formData.todo}
    onChange={changeHandler}
    />
    <button type="submit" className="add-btn">Add</button>
  </form>
  );
}

export default NewTodoForm;
