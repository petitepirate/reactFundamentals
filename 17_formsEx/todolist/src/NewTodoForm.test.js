import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it('renders without crashing', () => {
  render(<NewTodoForm formData={{
    todo: '',
    isComplete: false,
  }} />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm formData={{
    todo: '',
    isComplete: false,
  }} />)
  expect(asFragment()).toMatchSnapshot();
})

it("runs the create function on form submit", function() {
  const submitMock = jest.fn();
  const { getByText } = render(<NewTodoForm formData={{
    todo: '',
    isComplete: false,
  }} submitHandler={submitMock} />);
  const submitButton = getByText("Add");
  fireEvent.click(submitButton);
  expect(submitMock).toHaveBeenCalled();
});
