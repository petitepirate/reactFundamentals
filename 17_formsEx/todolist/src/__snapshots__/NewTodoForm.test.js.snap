// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Input items for your to-do list:
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button
      class="add-btn"
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
